### PR TITLE
Signed struct builder

### PR DESCRIPTION
**Pull request by mgrand**
This PR is to generate and use a builder to create instances of structs annotated with the 
`#[signed_struct]`
macro

### CODE REVIEW

File: pyrsia_node/src/docker/v2/handlers/manifests.rs [hunk 1]
```diff
@@ -297,49 +299,61 @@ fn package_version_from_schema1(
...
-            None,
-            None,
-            Map::new(),
-            None,
-        ))
+        add_fslayers(&mut artifacts, fslayer)?;
     }
-    Ok(PackageVersion::new(
-        String::from(
+    Ok(PackageVersionBuilder::default()
```
Comment (efrisch) [inline pyrsia_node/src/docker/v2/handlers/manifests.rs:304]: Much nicer looking!

**Review by efrisch — DISMISSED** [review]
Thanks for doing this Mark; I think it looks much cleaner. 👍

**Review by chb0github — APPROVED** [review]

**Comment by prince-chrismc** [comment]
is this relate too https://github.com/pyrsia/pyrsia/issues/182? If so can you please link them

**Comment by mgrand** [comment]
The original PR uncovered a bug in the signing (Pyrsia#183). At Sudhindra's request, the fix was included in this PR along with various documentation improvements.

File: src/signed_struct/src/lib.rs [hunk 2]
```diff
@@ -79,11 +105,13 @@ pub fn signed_struct(args: TokenStream, input: TokenStream) -> TokenStream {
                 }
                 let output = quote! {
                 #[derive(serde::Serialize, serde::Deserialize)]
```
Comment (efrisch) [inline src/signed_struct/src/lib.rs:107]: One derive instead of three?

**Review by efrisch — APPROVED** [review]
👍

**Review by chb0github — APPROVED** [review]

**Pull request merged**